src/librustc_driver/driver.rs

```diff
@@ -1066,6 +1064,13 @@ pub fn phase_3_run_analysis_passes<'tcx, F, R>(trans: &TransCrate,
             stability::check_unused_or_stable_features(tcx)
         });
 
+
+        time(time_passes,
+             "MIR linting",
+             || for def_id in tcx.body_owners() {
+                 mir::check_const_err::check(tcx, def_id)
```

Comment:
I think this should be triggered from one of the MIR queries, as per:
https://github.com/rust-lang/rust/blob/679f30e1aac510082b0fd3a2b5465321753aa7d1/src/librustc_mir/transform/mod.rs#L188-L189